ddp-workspace/projects/ddp-testboston/src/app/components/header/header.component.ts

```diff
@@ -21,10 +25,36 @@ export class HeaderComponent {
     return this.session.isAuthenticatedSession();
   }
 
+  public openMenu(): void {
+    this.isMenuOpened = true;
+    this.window.nativeWindow.requestAnimationFrame(() => {
+      this.renderer.addClass(this.menu.nativeElement, 'menu_visible');
+    });
+    this.window.nativeWindow.requestAnimationFrame(() => {
+      this.renderer.addClass(this.overlay.nativeElement, 'overlay_visible');
+    });
+  }
+
+  public closeMenu(): void {
+    if (this.isMenuOpened) {
+      this.isMenuOpened = false;
+      this.window.nativeWindow.requestAnimationFrame(() => {
+        this.renderer.removeClass(this.overlay.nativeElement, 'overlay_visible');
+      });
+      this.window.nativeWindow.requestAnimationFrame(() => {
+        this.renderer.removeClass(this.menu.nativeElement, 'menu_visible');
+      });
+    }
+  }
```

Comment:
This is a little bit tricky: to make header menu animation smooth we should change styles one by one in the different rendering cycles otherwise, the browser combines style changes in one rendering cycle and the header menu will "jump" without animation.